test/com/google/enterprise/adaptor/fs/WindowsFileDelegateTest.java

```diff
@@ -936,6 +937,113 @@ public boolean pushRecord(DocIdPusher.Record record) {
     delegate.destroy();
   }
 
+  @Test
+  public void testMonitorBackOffOnError() throws Exception {
+    Path file1 = newTempFile(tempRoot, "test1.txt");
+    Path file2 = newTempFile(tempRoot, "test2.txt");
+    Path file3 = newTempFile(tempRoot, "test3.txt");
+    byte[] contents = "Hello World".getBytes("UTF-8");
+
+    // Delegate with a Kernel32Ex that fails ReadDirectoryChangesW.
+    Kernel32Ex kernel32 =
+        new ChangeFailingKernel32(Kernel32Ex.INSTANCE, 0, 64, 64, 53);
+
+    WindowsFileDelegate delegate =
+        new WindowsFileDelegate(null, kernel32, null, null, 2000);
+
+    delegate.startMonitorPath(tempRoot, pusher);
+
+    Files.write(file1, contents);
+    checkForChanges(Collections.singleton(newRecord(file1)));
+
+    // This should be missed during the ReadDirectoryChangesW errors
+    Files.write(file2, contents);
+
+    // Wait for the error BackOffs to expire.
+    Thread.sleep(2500);
```

Comment:
Maybe note that the wait for the 3 configured errors is 0.5 + 0.75 + 1.125 = 2.375 < 2.5, unless I misunderstand what's happening here?